traits/trait_notifiers.py

```diff
@@ -27,27 +27,33 @@
 
 # Global Data
 
-# The thread ID for the user interface thread
-ui_thread = -1
+# The currently active handler for notifications that must be run on the UI
+# thread, or None if no handler has been set.
+_ui_handler = None
 
-# The handler for notifications that must be run on the UI thread
-ui_handler = None
+
+def get_ui_handler():
+    """
+    Return the current user interface thread handler.
+    """
+    return _ui_handler
 
 
 def set_ui_handler(handler):
     """ Sets up the user interface thread handler.
     """
-    global ui_handler, ui_thread
+    global _ui_handler
 
-    ui_handler = handler
-    ui_thread = threading.current_thread().ident
+    _ui_handler = handler
```

Comment:
Yes: TraitsUI already knows about and uses `set_ui_handler`. The `get_ui_handler` is really just for tests right now (mostly so that we can reset state at the end of a test that changes the handler), but it seems like an obvious counterpart to `set_ui_handler`.